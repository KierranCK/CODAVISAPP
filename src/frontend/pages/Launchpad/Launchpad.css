.launchpad {
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  position: relative;
}

.launchpad-title-pane {
  z-index: 10;
  height: 100%;
  display: -webkit-box;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-orient: vertical;
  -webkit-box-direction: normal;
      -ms-flex-direction: column;
          flex-direction: column;
  position: relative;
  left: -7.8em;
  padding-left: 5em;
  width: 100%;
  -webkit-transform: translateX(100px);
          transform: translateX(100px);
  height: 100vh;
  -webkit-box-pack: center;
      -ms-flex-pack: center;
          justify-content: center;
  gap: 0.8em;
}

.visualise-image {
  z-index: -1;
}

.visualise-image::before {
  content: "";
  background: -webkit-gradient(linear, left top, right top, from(transparent), color-stop(75%, black), to(transparent));
  background: linear-gradient(90deg, transparent, black 75%, transparent);
  position: absolute;
  height: 100vh;
  width: 50em;
  left: 10%;
  pointer-events: none;
  z-index: 10;
}

.visualise-image img {
  height: 100vh;
  z-index: -10;
  right: 0;
}

button {
  padding: 0.5em 2.5em !important;
}

.codavis-logo {
  width: 7em;
  position: absolute;
  top: -13em;
  left: -1em;
}

.codavis-logo-div {
  background: grey;
  position: relative;
}
/*# sourceMappingURL=Launchpad.css.map */