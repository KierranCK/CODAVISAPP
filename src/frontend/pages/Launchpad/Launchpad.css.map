{
    "version": 3,
    "mappings": "AAAA,AAAA,UAAU,CAAC;EACT,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,QAAQ,EAAE,QAAQ;CACnB;;AAED,AAAA,qBAAqB,CAAC;EACpB,OAAO,EAAE,EAAE;EACX,MAAM,EAAE,IAAI;EACZ,OAAO,EAAE,IAAI;EACb,cAAc,EAAE,MAAM;EACtB,QAAQ,EAAE,QAAQ;EAClB,IAAI,EAAE,MAAM;EACZ,YAAY,EAAE,GAAG;EACjB,KAAK,EAAE,IAAI;EACX,SAAS,EAAE,iBAAiB;EAC5B,MAAM,EAAE,KAAK;EACb,eAAe,EAAE,MAAM;EACvB,GAAG,EAAE,KAAK;CACX;;AAED,AAAA,gBAAgB,CAAC;EACf,OAAO,EAAE,EAAE;CAgBZ;;AAjBD,AAEE,gBAFc,AAEb,QAAQ,CAAC;EACR,OAAO,EAAE,EAAE;EACb,UAAU,EAAE,2DAKX;EACC,QAAQ,EAAE,QAAQ;EAClB,MAAM,EAAE,KAAK;EACb,KAAK,EAAE,IAAI;EACX,IAAI,EAAE,GAAG;EACT,cAAc,EAAE,IAAI;EACpB,OAAO,EAAE,EAAE;CACZ;;AAGH,AAAA,gBAAgB,CAAC,GAAG,CAAC;EACnB,MAAM,EAAE,KAAK;EACb,OAAO,EAAE,GAAG;EACZ,KAAK,EAAE,CAAC;CACT;;AAED,AAAA,MAAM,CAAC;EACL,OAAO,EAAE,sBAAsB;CAChC",
    "sources": [
        "Launchpad.scss"
    ],
    "names": [],
    "file": "Launchpad.css"
}