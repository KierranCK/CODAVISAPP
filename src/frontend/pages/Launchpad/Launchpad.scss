.launchpad {
  height: 100%;
  display: flex;
  position: relative;
}

.launchpad-title-pane {
  z-index: 10;
  height: 100%;
  display: flex;
  flex-direction: column;
  position: relative;
  left: -7.8em;
  padding-left: 5em;
  width: 100%;
  transform: translateX(100px);
  height: 100vh;
  justify-content: center;
  gap: 0.8em;

  &::after {
    content: "";
  background: linear-gradient(
    90deg,
    transparent,
    rgb(0, 0, 0) 75%,
    transparent
  );
    position: absolute;
    height: 100vh;
    width: 50em;
    right: -150px;
    pointer-events: none;
  }
}

.visualise-image img {
  height: 100vh;
  z-index: -10;
  right: 0;
}

button {
  padding: 0.5em 2.5em !important;
}
