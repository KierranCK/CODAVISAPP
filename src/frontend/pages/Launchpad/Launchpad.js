import React from 'react'
import Button from '../../components/Button/Button'
import visualise from '../../images/visualise-full-cropped-min.jpg'
import './Launchpad.css'

const Launchpad = () => {
  return (
      <div className='launchpad'>

      <div className='launchpad-title-pane'>
        <h1>CODVID DATA <br /> VISUALISATION</h1>
        <p>A powerful tool for the analysis and visualisation of COVID-19</p>
        <Button 
        location="/dashboard"
        label="Launch"
        colour="orange"
        />
        </div>

        <div className='visualise-image'>
            <img src={visualise} />
        </div>

    </div>
  )
}

export default Launchpad