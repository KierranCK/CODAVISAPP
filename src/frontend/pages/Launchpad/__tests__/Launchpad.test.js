import { render, screen, cleanup } from "@testing-library/react";
import React from "react";
import { BrowserRouter } from "react-router-dom";
import Launchpad from "../Launchpad";

//function to render component to be tested
const renderComponent = () =>
  render(<Launchpad />, {
    wrapper: BrowserRouter,
  });

//object containing all elements to be tested
const elements = {
  heading: () => screen.getByRole("heading"),
  button: () => screen.getByRole("button"),
  image: () => screen.getByRole("img"),
};

//function to render component before each test
beforeEach(() => {
  renderComponent();
});

//function cleanup after each test
afterEach(() => {
  cleanup();
});

//test suite
describe("Launchpad Tests", () => {
  test("Ensure Launchpad has not inadvertently changed", () => {
    expect(renderComponent().asFragment()).toMatchSnapshot();
  });
  test('Ensure Title, image and "Launch" button rendered', () => {
    expect(elements.heading()).toHaveTextContent("COVID");
    expect(elements.button()).toBeInTheDocument();
    expect(elements.button()).toHaveTextContent("Launch");
    expect(elements.button()).not.toBeDisabled();
    expect(elements.image()).toBeInTheDocument();
  });
});
