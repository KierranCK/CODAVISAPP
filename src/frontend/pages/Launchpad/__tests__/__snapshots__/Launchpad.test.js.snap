// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Launchpad Tests Ensure Launchpad has not inadvertently changed 1`] = `
<DocumentFragment>
  <div
    class="launchpad"
  >
    <div
      class="launchpad-title-pane"
    >
      <h1>
        COVID DATA 
        <br />
         VISUALISATION
      </h1>
      <p>
        A powerful tool for the analysis and visualisation of COVID-19
      </p>
      <div
        class="row"
      >
        <div
          class="col"
        >
          <a
            href="/dashboard"
          >
            <button
              class="btn btn-warning btn-sm"
              type="button"
            >
              Launch
            </button>
          </a>
        </div>
      </div>
    </div>
    <div
      class="visualise-image"
    >
      <img
        alt="visualise-image"
        src="visualise-full-cropped-min.jpg"
      />
    </div>
  </div>
</DocumentFragment>
`;
